streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBuffer.java

```diff
@@ -45,17 +53,24 @@ public class RocksDBTimeOrderedKeyValueBuffer<K, V> extends WrappedStateStore<Ro
     private Serde<V> valueSerde;
     private final String topic;
     private int seqnum;
+    private final boolean loggingEnabled;
+    private final Map<Bytes, BufferValue> dirtyKeys = new HashMap<>();
```

Comment:
Why do we introduce this dirtyKeys buffer? If we buffer KStream key-value pairs, key must be unique, because we cannot overwrite records with the same key. Thus, thus Map would only grow. Why do we need to buffer anything to begin with? Seems we could just call `logValue` / `logTombstone` directly instead of waiting for `flush()`?